### PR TITLE
chore: bump alloy-eip7702

### PR DESCRIPTION
**Pull request by klkvr**
alloy-eip7702 now has logic for authority recovery implemented as it should happen in EVM https://github.com/alloy-rs/eips/blob/e712799f4318634bc978d0749cfcd59ff9a59459/crates/eip7702/src/auth_list.rs#L234

This PR updates it to latest version and replaces `RecoveredAuthority` type with one from alloy-eip7702.

There's also now no need for authority list validity check as this is done on decoding

### CODE REVIEW

**Comment by klkvr** [comment]
@rakita tests are failing on release/v47 branch even before my commits, so I guess this is fine?

**Comment by rakita** [comment]
> @rakita tests are failing on release/v47 branch even before my commits, so I guess this is fine?

Let me check what is going on, i didn't check it previously as it was just a primitives bump

**Review by rakita — APPROVED** [review]
On second look, this look good

File: crates/revm/src/handler/mainnet/pre_execution.rs [hunk 1]
```diff
@@ -115,9 +115,8 @@ pub fn apply_eip7702_auth_list<SPEC: Spec, EXT, DB: Database>(
     let mut refunded_accounts = 0;
     for authorization in authorization_list.recovered_iter() {
         // 1. Verify the chain id is either 0 or the chain's current ID.
-        if !authorization.chain_id() == 0
```
Comment (rakita) [inline crates/revm/src/handler/mainnet/pre_execution.rs:118]: There was a bug here, `!chain_Id` would be a bitwise operation.

**Pull request merged**